src/ipaContent.ts

```diff
@@ -1,70 +1,58 @@
+import { endsWith } from 'lodash';
+import { Utils } from './utils';
 
 export type ProfileType = "adhoc" | "enterprise" | "other";
 
-export class ProvisioningProfile {
+export class ProvisioningProfile implements IProvisioningProfile {
     idName: string;
     name: string;
     teamIdentifier: string;
     profileType: ProfileType;
-    expiredAt: string;
+    expiredAt: Date;
     mobileProvisionFileContent: string;
     UniqueDeviceIdentifierList: string;
     pathName: string;
 }
 
-export class IpaContent extends ContentBase implements IIpaContent {
+export class IpaContent extends ContentBase implements IIPAMetadata {
     provision: ProvisioningProfile;
     appexProvisioningProfiles: ProvisioningProfile[];
 
     public get supportedFiles(): string[] {
         return Constants.IOS_FILES;
     }
-    public async read(tempDir: string, fileList: any): Promise<any> {
+    public async read(tempDir: string, fileList: any): Promise<void> {
         this.provision = new ProvisioningProfile();
         const plistData = await this.parsePlist(fileList, tempDir);
         this.iconFullPath = await this.parseIcon(fileList, tempDir, plistData);
         this.mapPlist(plistData, this.iconFullPath);
         this.parseLanguages(fileList);
         const provisionData = await this.parseProvision(this.provision, Constants.PROVISIONING, tempDir, fileList);
-        this.mapProvision(this. provision, provisionData);
+        this.mapProvision(this.provision, provisionData);
         await this.parseAppex(fileList, tempDir);
-        return this;    
     }
     private iconSearch(fileList: string[]): string {
-        let chosenIcon = null;
-        for (const plistIcon of fileList) {
-            if (!chosenIcon && plistIcon.toLowerCase().includes("png")) {
-                chosenIcon = plistIcon;
-            }
-            if (plistIcon.includes("3x")) {
-                if (plistIcon.includes("ipad") && !chosenIcon.includes("ipad")) {
-                    chosenIcon = plistIcon;
-                    return chosenIcon;
-                }
-                if (!chosenIcon.includes("3x")) {
-                    chosenIcon = plistIcon;
-                }
-            }
-            if (plistIcon.includes("2x")) {
-                if (!chosenIcon.includes("3x") && !chosenIcon.includes("2x")) {
-                    chosenIcon = plistIcon;
-                } else if (chosenIcon.includes("2x")) {
-                    if (plistIcon.includes("ipad") && !chosenIcon.includes("ipad")) {
-                        chosenIcon = plistIcon;
-                    }
-                }
-            }
-        }
-        return chosenIcon;
+        // getting the best resolution image
+        const pngFiles = fileList.filter(value => value && endsWith(value.toLowerCase(), '.png'));
+        if (pngFiles.length === 0) {
+            return null;
+        }
+        return Utils.findFileByPrioritizedPatterns(pngFiles, [
+            ['ipad', '@3x'],
```

Comment:
the 60x60 3x (total 180x180 pixels) image is the highest resolution one. iPad doesn't support a 3x retina display yet.

Not all of the app icon assets might be included, so this won't get the highest resolution image all of the time. I think it would be better to get the resolution straight from the file info, it's possible through a variety of ways: https://stackoverflow.com/questions/12539918/get-the-width-and-height-of-an-image-in-node-js

![image](https://user-images.githubusercontent.com/3309218/34394451-d8642ff0-eb1e-11e7-9dfd-0f2183936ea5.png)

Reply:
Yes, you are right. For now I just simplified the code, we can make improvement as needed. 
I will make this comment a GitHub issue.